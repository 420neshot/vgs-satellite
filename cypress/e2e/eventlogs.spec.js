describe('Localhoste eventlogs', function() {
  beforeEach(() => {
    cy.server();
    cy.route('GET', 'flows.json').as('getFlows');
    cy.cleanupFlows();
    cy.fixCypressSpec(__filename);
  });

  it('Check eventlogs tab', function() {
    cy.visit('/');
    cy.wait(['@getFlows']);
    cy.get('[data-role="demo-curl"]').toMatchSnapshot();

    cy.task('viaProxy', {
      method: 'POST',
      url: 'http://httpbin.org/post',
      data: { foo: 'bar' },
    });

    cy.get('[data-role="logs-row"]').should('have.length', 1);
    cy.get('[data-role="logs-row"] .badge-success').contains('200');
    cy.get('[data-role="logs-row"]').contains('/post').click();

    cy.get('[data-role="tab-events"]').click();

    cy.get('.eventlogs-table [data-role="event-name"]')
      .each(($el) => {
        cy.wrap($el).toMatchSnapshot();
        cy.wrap($el).click();
        cy.wrap($el).parents('.ant-table-row-level-0').next().children().toMatchSnapshot();
        cy.wrap($el).click();
      });
  });
});
