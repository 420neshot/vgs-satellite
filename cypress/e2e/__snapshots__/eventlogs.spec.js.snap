exports[`Localhoste eventlogs > Check eventlogs tab #0`] = `
<div data-role="demo-curl">
  <pre
    class="line-numbers ant-card card px-5 bg-light  language-bash"
  ><code class="  language-bash"><span class="token function">curl</span> https://httpbin.org/post -k <span class="token punctuation">\\</span>
  -x localhost:9099 <span class="token punctuation">\\</span>
  -H <span class="token string">"Content-type: application/json"</span> <span class="token punctuation">\\</span>
  -d <span class="token string">'{"foo": "bar"}'</span></code><button type="button" class="position-absolute top right mt-2 text-muted btn btn-link btn-sm"><span id=""><i class="fa-copy fas mr-2 text-text"></i></span>Copy</button></pre>
</div>
`;

exports[`Localhoste eventlogs > Check eventlogs tab #1`] = `
<span class="_600 text-text-dark" data-role="event-name">Proxy request</span>
`;

exports[`Localhoste eventlogs > Check eventlogs tab #2`] = `
<td class="" colspan="2">
  <div data-role="event-details">
    <p class="mb-0 ml-4 pl-1 py-2 small-capsy" data-role="method">
      <span class="text-text-light pr-2" data-role="name">Method</span
      ><span data-role="value">POST</span>
    </p>
    <p class="mb-0 ml-4 pl-1 py-2 small-capsy" data-role="uri">
      <span class="text-text-light pr-2" data-role="name">Uri</span
      ><span data-role="value">http://httpbin.org/post</span>
    </p>
  </div>
</td>
`;

exports[`Localhoste eventlogs > Check eventlogs tab #3`] = `
<span class="_600 text-text-dark" data-role="event-name">Proxy traffic</span>
`;

exports[`Localhoste eventlogs > Check eventlogs tab #4`] = `
<span class="_600 text-text-dark" data-role="event-name">Proxy traffic</span>
`;

exports[`Localhoste eventlogs > Check eventlogs tab #5`] = `
<span class="_600 text-text-dark" data-role="event-name"
  >Upstream response</span
>
`;

exports[`Localhoste eventlogs > Check eventlogs tab #6`] = `
<td class="" colspan="2">
  <div data-role="event-details">
    <p class="mb-0 ml-4 pl-1 py-2 small-capsy" data-role="upstream">
      <span class="text-text-light pr-2" data-role="name">Upstream</span
      ><span data-role="value">httpbin.org</span>
    </p>
    <p class="mb-0 ml-4 pl-1 py-2 small-capsy" data-role="status_code">
      <span class="text-text-light pr-2" data-role="name">Status code</span
      ><span data-role="value">200</span>
    </p>
  </div>
</td>
`;
